### PR TITLE
add 'fp32' option for softmax_mode in FusedFSDPA

### PR DESCRIPTION
**Pull request by yangulei**
The 'fp32' option for `softmax_mode` in [FusedFSDPA](https://docs.habana.ai/en/v1.20.0/PyTorch/Reference/Python_Packages.html#hpex-kernels-fusedsdpa) is enabled since Synapse v1.20.0.

### CODE REVIEW

**Review by michalkuligowski — APPROVED** [review]

**Comment by yangulei** [comment]
Update:
[set valid_seq_lengths=None for softmax_mode='fp32'](https://github.com/HabanaAI/vllm-hpu-extension/pull/113/commits/04ea048b1f7b2a0ac3a31fd935d3e782c07767b9) as we observed [accuracy issue on Qwen2-1.5B-Instruct and Deepseek-R1](https://habana.atlassian.net/jira/servicedesk/projects/CS/queues/custom/94/CS-1143) for `softmax_mod='fp32'` and setting `valid_seq_lengths=None` could fix it.

**Review by michalkuligowski — COMMENTED** [review]
@yangulei Is this still valid, given your last comment?

**Comment by yangulei** [comment]
> @yangulei Is this still valid, given your last comment?

Yes, no more accuracy issue observed after setting `valid_seq_lengths=None` for `softmax_mod='fp32'` case.

**Comment by michalkuligowski** [comment]
@yangulei there is conflict in _naive_prompt_attention method, can you please fix?

**Comment by yangulei** [comment]
Close as already included in latest code.
